Code/Framework/AzCore/AzCore/Serialization/Json/CastingHelpers.h

```diff
@@ -24,9 +24,9 @@ namespace AZ
     {
         using namespace JsonSerializationResult;
 
-        if (NumericCastInternal::FitsInToType<ToType>(value))
+        if (NumericCastInternal::template FitsInToType<ToType>(value))
         {
-            result = aznumeric_cast<ToType>(value);
+            result = static_cast<ToType>(value);
```

Comment:
here the check is already done by the `FitsInToType` so there is no need to do it again. I can leave it as an `aznumeric_cast` if preferred